internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -205,6 +222,20 @@ func (h *Diagnostics) runHooks(ctx context.Context) ([]client.DiagnosticFileResu
 		elapsed := time.Since(startTime)
 		h.log.Debugw(fmt.Sprintf("Hook %s execution complete, took %s", hook.Name, elapsed.String()), "hook", hook.Name, "filename", hook.Filename, "elapsed", elapsed.String())
 	}
+	if collectCPU {
+		p, err := diagnostics.CreateCPUProfile(ctx, 30*time.Second)
```

Comment:
We should probably define 30s as a constant so it can be updated in both of the places its used

https://github.com/elastic/elastic-agent/blob/8ea0d04402aa62cf745d39ff535bc4ce7f642e23/pkg/control/v2/server/server.go#L208